libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/NotificationToEmailConverterCommon.java

```diff
@@ -69,7 +72,26 @@ public NotificationToEmailConverterCommon() {
         supportUrl = System.getProperty(PROP_NOTIFICATION_SUPPORT_URL);
 
         emailBaseCSS = readContentFromFile(getClass().getClassLoader(), EMAIL_TEMPLATE_CSS);
+        this.notificationUserAuthority = loadNotificationUserAuthority();
+    }
+
+    private Authority loadNotificationUserAuthority() {
```

Comment:
Was there any reason why we're not just passing the user authority object from the server to the notification interface so we don't have to maintain a separate setting just for the notification library and then load it again?

Reply:
There were a couple of reasons I decided to go this route:
1. For ZTS we will need a new setting in any case as we currently don't load a "UserAuthority" there (we load all the authorities but we don't keep a specific one for users like we do in ZMS).
2. I thought it might be possible that we will want to use a separate authority for Notifications than the one we usually use for authentication.
If you think it's best to inject the existing user authority let me know and I'll modify the "loadAuthorities" method in ZTSImpl to load it and later inject it there and from zms.

Reply:
My concern is that I might have some cases where I don't want to instantiate multiple copies of my user authority class. Those could be doing some heavy operations to get the data, etc, so now we're doing twice the work.  So if there is a use case to have a different user authority then the implementation could be to look at the current setting - if one is defined then it will load that otherwise it should use whatever the server gave it - this way we're only using a single instance.

As far ZTS - we can just introduce the setting in ZTS and then pass the authority to the notification library.

Reply:
Sure I'll make the necessary changes.